includes/installer/class-installer.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * File containing the class \Sensei\Installer\Installer.
+ *
+ * @package sensei
+ * @since $$next-version$$
+ */
+
+namespace Sensei\Installer;
+
+use Sensei_Updates;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Installer class.
+ *
+ * Responsible for running DB updates that need to be run per version.
+ *
+ * @since $$next-version$$
+ */
+class Installer {
+
+	private const SENSEI_VERSION_OPTION_NAME = 'sensei-version';
+
+	/**
+	 * Instance of the class.
+	 *
+	 * @since $$next-version$$
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * The database schema class.
+	 *
+	 * @since $$next-version$$
+	 * @var Schema
+	 */
+	private $schema;
+
+	/**
+	 * Current Sensei version.
+	 *
+	 * @var string|null
+	 */
+	private $version;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param Schema      $schema Schema migration object.
+	 * @param string|null $version Current Sensei version.
+	 */
+	public function __construct( Schema $schema, ?string $version ) {
+		$this->schema  = $schema;
+		$this->version = $version;
+	}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string|null $version Current Sensei version.
+	 * @return self
+	 */
+	public static function instance( ?string $version = null ): self {
+		if ( ! self::$instance ) {
+			self::$instance = new self( new Schema(), $version );
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initialize necessary hooks.
+	 *
+	 * @since $$next-version$$
+	 */
+	public function init() {
+		register_activation_hook( SENSEI_LMS_PLUGIN_FILE, [ $this, 'install' ] );
+		add_action( 'plugins_loaded', [ $this, 'install' ] );
+		add_action( 'init', array( $this, 'update' ) );
+	}
+
+	/**
+	 * Run the installer.
+	 *
+	 * This method is executed when the plugin is installed or updated.
+	 *
+	 * @since $$next-version$$
+	 */
+	public function install() {
+		if (
+			! is_blog_installed()
+			|| $this->is_installing()
+			|| ! $this->requires_install()
+		) {
+			return;
+		}
+
+		set_transient( 'sensei_lms_installing', 'yes', MINUTE_IN_SECONDS * 10 );
+
+		$this->schema->create_tables();
+
+		delete_transient( 'sensei_lms_installing' );
+
+		/**
+		 * Fires after the installation completes.
+		 *
+		 * @since $$next-version$$
+		 */
+		do_action( 'sensei_lms_installed' );
+	}
+
+	/**
+	 * Checks for plugin update tasks and ensures the current version is set.
+	 *
+	 * @since $$next-version$$
+	 */
+	public function update(): void {
+		$current_version = get_option( self::SENSEI_VERSION_OPTION_NAME );
+		$is_upgrade      = $current_version && version_compare( $this->version, $current_version, '>' );
+
+		// Make sure the current version is up-to-date.
+		if ( ! $current_version || $is_upgrade ) {
+			$this->update_version();
+		}
+
+		$updates = Updates_Factory::create( $current_version, $is_upgrade );
+		$updates->run_updates();
+	}
```

Comment:
Could you cover this method with tests as well?

Reply:
I think these tests cover it. But I'll move them to the proper place 👌